library/cpp/threading/local_executor/tbb_local_executor.cpp

```diff
@@ -0,0 +1,39 @@
+#include "tbb_local_executor.h"
+
+void NPar::TTbbLocalExecutor::SubmitAsyncTasks(TLocallyExecutableFunction exec, int firstId, int lastId) {
+    for (int i = firstId; i < lastId; ++i) {
+        Group.run([=] { exec(i); });
+    }
+}
+
+int NPar::TTbbLocalExecutor::GetThreadCount() const noexcept {
+    return NumberOfTbbThreads - 1;
+}
+
+int NPar::TTbbLocalExecutor::GetWorkerThreadId() const noexcept {
+    return  tbb::this_task_arena::current_thread_index();
+}
+
+void NPar::TTbbLocalExecutor::Exec(TIntrusivePtr<ILocallyExecutable> exec, int id, int flags) {
+    if (flags & WAIT_COMPLETE) {
+        exec->LocalExec(id);
+    } else {
+        TbbArena.execute([=] {
+            SubmitAsyncTasks([=] (int id) { exec->LocalExec(id); } , id, id + 1);
+        });
+    }
+}
+
+void NPar::TTbbLocalExecutor::ExecRange(TIntrusivePtr<ILocallyExecutable> exec, int firstId, int lastId, int flags) {
+    if (flags & WAIT_COMPLETE) {
+        TbbArena.execute([=] {
+            tbb::this_task_arena::isolate([=]{
+                tbb::parallel_for(firstId, lastId, [=] (int id) { exec->LocalExec(id); });
+            });
```

Comment:
let us add private bool RespectTls = false to TTbbLocalExecutor, and add isolate here, only if RespectTls == true
this is needed to preserve current performance on datasets with cat features
```c++
TTbbLocalExecutor(int nThreads, bool respectTls = false)
    : ...
    , RespectTls(respectTls)
{
}
```

Reply:
done